pkg/capture/capture.go

```diff
@@ -377,14 +382,39 @@ func (c *Capture) status() (*capturetypes.CaptureStats, error) {
 	return &res, nil
 }
 
+func (c *Capture) fetchStatusInBackground(ctx context.Context) (res chan *capturetypes.CaptureStats) {
```

Comment:
Getting the status can take a moment, but is independent from the actual rotation (since the capture is locked there is no potential race with the counters), so we can do it in the background _while_ doing the rotation. Depending on how long it actually takes this can save some time during which buffering is required.